src/WebJobs.Script/ServiceManagers/IManagedHostedService.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Script.ServiceManagers
+{
+    public interface IManagedHostedService
```

Comment:
IStopperService? Also, what is the motivation behind making this IHostedService..?

Reply:
That doesn’t really follow naming conventions. Maybe `ISupportOuterScopeStop` or something along those lines. The motivation for it implementing `IHostedSercice` is the name, if it stays as is. It would be confusing to call this a hosted service if it isn’t one. Implementing the interface would fix that

Reply:
This now extends IHostedService and any service that extends IManagedHostedService should explicitly implement both StopAsync() in IManagedHostedService and IHostedService